WordPress/gutenberg
# [RNMobile] Add inserter block search

## Description
Adds a block search form to the existing search menu.  The available blocks are filtered according to the search query. 
The search form is only rendered when a minimal number of blocks are available in the menu (the current min is set to 2) 

The new feature is behind the `__DEV__` flag in this change set.

Part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/2570

## How has this been tested?



### Testing search flow
- Open the mobile block inserter
- Press "Search blocks"
- Try searching for a block with the soft keyboard
   - The inserter menu should filter the blocks per the search string
- Try pressing on a block after searching
   - The chosen block should be inserted in the expected position 
   - The inserter menu should close

- Press "Search blocks"
    - The prior search input should be cleared 
- Try searching with a single character i.e, `i`
   - The blocks in the menu should be sorted to best match the search
   - The menu should be scrollable   
   
### Testing canceling search flows
- Open the mobile block inserter
- Press "Search blocks"
- Press the arrow on the left side of the input
    - The inserter menu should return to the original inserter menu

- Press "Search blocks" again
- Start searching for a block
- Press the x on the right side of the input
    - The input field should clear and the inserter menu should show all available blocks 

- Start searching for a block again
- Press the arrow on the left side of the input
- Press "Search blocks" again
    - The input field should reset to the placeholder text  

### Testing with few available blocks
- Add a Buttons block
- With the Buttons block selected, open the block inserter
    - The search form should not be visible  

## Screenshots 

![inseter-block-search mp4](https://user-images.githubusercontent.com/744755/108565343-5f667b00-72b9-11eb-97bd-00603e845513.gif)


## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
